src/automation/tasks/semantic_similarity_generate.py

```diff
@@ -0,0 +1,121 @@
+from automation.tasks.base_task import BaseTask
+from automation.configs import DEFAULT_DOCKER_IMAGE
+from typing import Union, List, Optional, Sequence, Any, Callable
+import os
+import yaml
+
+class SemanticSimilarityGenerateTask(BaseTask):
+    task_packages = [
+        "vllm",
+        "hf_xet",
+        "pyzmq",
+        #"vllm==0.10.1.1",
+    ]
+
+    def __init__(
+        self,
+        project_name: str,
+        task_name: str,
+        model_id: str,
+        branch: str,
+        max_new_tokens: int,
+        max_model_len: int,
+        num_samples_per_dataset: Optional[int],
```

Comment:
Instead of num_samples_per_dataset we should enable the user to define datasets and number of samples for each separately. Maybe something like:
datasets = {"tatsu-lab/alpaca": 300, "garage-bAInd/Open-Platypus": 200}